src/main/java/com/redhat/red/build/finder/BuildFinder.java

```diff
@@ -562,7 +579,8 @@ public static void main(String[] args) {
                 LOGGER.info("Attempting to load existing builds file {}", buildsFile.getAbsolutePath());
                 builds = JSONUtils.loadBuildsFile(buildsFile);
             } else {
-                builds = findBuilds(checksums);
+                BuildFinder bf = new BuildFinder(session, config);
+                builds = bf.findBuilds(checksums);
                 JSONUtils.dumpFile(buildsFile, builds);
             }
```

Comment:
+1 ; potentially in future I suppose it could be refactored so main() just dealt with the incoming arguments and left all processing / reporting output to the 'object' which would make it easier to unit test.

Reply:
Based on what you said, yes, it makes sense to move the report handling to `BuildFinder` as well.